### PR TITLE
Adding a new Status API to Open MCT

### PR DESCRIPTION
**Pull request by deeptailor**
Adding a new Status API to allow developers to add statuses to domainObjects. Use cases at the moment include 

1. Default Notebook
2. Missing Object
3. Suspect DomainObjects

Future use cases could include flagging domainObjects as favorites etc

### CODE REVIEW

File: src/api/status/StatusAPISpec.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+            statusAPI.delete(identifier);
+            resultingStatus = statusAPI.get(identifier);
+
+            expect(resultingStatus).toBeUndefined();
+        });
+    });
+
+    describe("observe function", () => {
+
```
Comment (davetsay) [inline src/api/status/StatusAPISpec.js:63]: remove newline

**Comment by davetsay** [comment]
Awesome work @deeptailor

**Review by akhenry — CHANGES_REQUESTED** [review]
@deeptailor I think this looks really great, a nice clean approach which solves a bunch of problems.

I do think that objects are going to need to have multiple statuses simultaneously, (eg. a notebook could be both the default notebook and a user favorite, so they can find it quickly).

Perhaps we could change `get` and `set` so that it accepts a key and value?

For convenience we probably also want observers to be able to only listen to a particular status (eg. tell me when this notebook becomes the default notebook).

**Comment by akhenry** [comment]
@deeptailor I think we need to address the need for multiple statuses ASAP. eg. A notebook can be both the default notebook and be missing.

**Pull request merged**